qiskit/circuit/parametertable.py

```diff
@@ -165,7 +166,16 @@ def get_names(self):
         Returns:
             set: A set of all the names in the parameter table
         """
-        return self._names
+        return self._names.keys()
+
+    def parameter_from_name(self, name: str, default: typing.Any = None):
```

Comment:
Would it make sense to restrict the default to `Parameter | None`? Since `Parameter` would be the only type that would be returned by this function 🤔

Reply:
This is give-or-take a wrapper around `dict[K, V].get`, which has the signature `(K, T) -> V | T`, where `T = Optional[U]` for all `U`, not restricted to `U <: V`, so I was give-or-take just mirroring that.  It's largely unimportant because it's not a public method (`ParameterTable` is an internal-only detail), but do you think there's an improvement to be had by limiting what a user can do?  This function won't ever _do_ anything with the default, so as far as the function's concerned, it's a black-box object.

Are you worried about the type checking of the return value?  If I'd had Python 3.12's annotation scopes I'd have done
```python
@overload
def parameter_from_name[T](self, name: str, default: T) -> Parameter | T: ...
@overload
def parameter_from_name(self, name: str, default=None) -> Optional[Parameter]: ...
```
but I think I just got discouraged when I realised I'd have to make a `TypeVar` to spell it and didn't bother.  If that's the main problem, I'd happy to write it properly, then somebody feeding a `Parameter` in as the default will get the best type checking.

Reply:
Yeah I was mainly thinking about the return value, but it's true that it's not a user-facing method so I'm good leaving it as is 🙂